packages/web/src/References.tsx

```diff
@@ -0,0 +1,96 @@
+import * as Markprompt from '@markprompt/react';
+import React, { useCallback, useContext, useMemo } from 'react';
+
+const capitalize = (text: string) => {
+  return text.charAt(0).toUpperCase() + text.slice(1);
+};
+
+const removeFileExtension = (fileName: string) => {
```

Comment:
Should be part of `@markprompt/core`, or some shared utility package (holds for all utility functions found here which are also in use in `@markprompt/react`.

Reply:
I don't think these functions should be in `@markprompt/core`: 

- They are only used here (not in `@markprompt/react`)
- Users are likely to pass their own `transformReferenceId` function to construct reference texts and href's.
- There is not much 'core' about these functions, as
- They are just default props for the `References` component and users are likely to override them
- They are used in `with-css-modules` for the sake of the example of building a Markprompt dialog using our headless components, but I don't think they should be regarded as something reusable.

Reply:
You are right. I'm just concerned about the mental overhead to preserve behavior across example implementations. Not sure how we can best mitigate that, but indeed it doesn't belong to core.